### PR TITLE
refactor(compiler): don’t rely on external css parser

### PR DESCRIPTION
**Pull request by tbosch**
We used to use different external css parsers,
depending on the `DomAdapter`. This lead to
inconsistent behavior and environment specific errors.

Related to #4993

### CODE REVIEW

File: modules/angular2/src/core/compiler/shadow_css.ts [hunk 1]
```diff
@@ -347,50 +344,33 @@ export class ShadowCss {
...
+    for (var i = 0; i < parts.length; i += 2) {
+      var selectorText = parts[i];
+      var ruleContent = parts[i + 1];
+      var selectorMatch = RegExpWrapper.firstMatch(_singleSelectorRe, selectorText);
+      if (isPresent(selectorMatch) && ruleContent.length > 0) {
+        var selPrefix = selectorMatch[1];
+        var selAt = isPresent(selectorMatch[2]) ? selectorMatch[2] : '';
+        var selector = selectorMatch[3];
+        var selSuffix = selectorMatch[4];
+        if (selAt.length === 0 || selAt == '@page') {
```
Comment (tbosch) [inline modules/angular2/src/core/compiler/shadow_css.ts]: How about this case:
`@import 'asdf'; div {...}`?

**Comment by tbosch** [comment]
Test against acx demo in G3 first before merging!

File: modules/angular2/src/core/compiler/shadow_css.ts [hunk 2]
```diff
@@ -539,13 +489,35 @@ var _polyfillHostRe = RegExpWrapper.create(_polyfillHost, 'im');
...
-function _cssToRules(cssText: string) {
-  return DOM.cssToRules(cssText);
+var _singleSelectorRe = /^(\s*)(@\S+)?(.*?)(\s*)$/g;
+
+var _curlyRe = /([{}])/g;
+var OPEN_CURLY = '{';
+var CLOSE_CURLY = '}';
+
+export function splitCssRules(cssText:string):string[] {
+  var parts = StringWrapper.split(cssText, _curlyRe);
```
Comment (vicb) [inline modules/angular2/src/core/compiler/shadow_css.ts]: do we strip comment first ? (what if they contain curly brackets)

Comment (tbosch) [inline modules/angular2/src/core/compiler/shadow_css.ts]: We have the same problem with all the other regex in this file...

**Comment by tbosch** [comment]
Just verified against the ACX demo that this works.
Also fixes #4993

**Pull request closed without merging**